### PR TITLE
Phase 1: Activate Formal Languages Chapter

### PR DESCRIPTION
**Pull request by eAlasdair**
This PR activates the first three sections of the `Formal Languages` chapter:
- The introduction
- `Getting started`
- `Finite state automata`

This and Phase 2 (#897) can be merged in any order, conflicts will need to be resolved when merging in the second either way.

## Phase 1:
This is everything to do with the start of the chapter through `Finite state automata` section.

### Checks:
- [x] All related interactives work properly:
  - [x] `Regular Expression Filter`
  - [x] `Trainsylvania`
  - [x] `Trainsylvania Planner`
  - [x] `FSA Box`
  - [x] `FSA Washing Machine`
  - [x] `FSA Light`
- [x] Full internationalization support
- [x] Properly documented

### Changes:
- [x] Content fixes
- [x] Move the tutorials on Exorciser and JFLAP to 'project'-style panels
- [x] Rename interactive .css files to .scss files
- [x] Create uninteractives or otherwise edit images for internationalization:
  - [x] `python-syntax-error.png` **(Uninteractive)**
  - [x] `trainsylvania-blank.png` **(Uninteractive)**
  - [x] `trainsylvania-complete.png` **(Uninteractive)**
  - [x] `finite-state-automata-spin-speeds.png` **(edit)**
  - [x] `finite-state-automata-train-example-terminology.png` **(Uninteractive)**
  - [x] `finite-state-automata-light-switch-example.png` **(edit)**
  - [x] `finite-state-automata-microwave-example.png` **(edit)**
  - [x] `finite-state-automata-binary-multiples.png` **(edit)**
- [x] [12/12] Introduce solutions to FSA 'Examples to try'
- [x] Introduce proper licence files (Relates to #451)
- [x] Remove now unused files and images
- [x] Fix NFA Guesser from the 'The whole story!' section

### Notes:
- Notes that I make here regularly disappear, so it is most likely that this is not everything I wrote down.
- The `Regular Expression Filter` interactive only loads English text to filter through - according to the README this is purposeful
- Users can click a button to load the original `trainsylvania-blank` image in PDF form to allow them to print it. In this the station names remain in English and (as far as I know) don't get translated for other languages.
- The `trainsylvania-blank` & `trainsylvania-complete` uninteractives source their images from the `trainsylvania` interactive img directory
- `trainsylvania-complete` is small because it is too much effort for the result to rebuild the layout in different sizes. It also loads in whole-page mode for this reason and will load its thumbnail (the original image) instead if the screen is less than 500px wide
- The SVG canvas for `trainsylvania-complete` is created below the icons and shifted up. As such the uninteractive is twice as tall as it appears. This is noticeable in that its thumbnail looks squashed, but is otherwise not so
- Two of the given FSA challenges (weekday and month) can't actually be completed with Exorciser. I encountered a bug where it stops working when too many unique characters are used in the alphabet. Instead I used this site, http://www.madebyevan.com/fsm/, to create the solutions.
- The paper sizes FSA challenge is both ambiguous and, in its most likely format, very similar to the month numbers FSA. It has been removed but if I think of something better I aim to replace it.
- The XKCD comics used remain in English
- The CSU Treasure Hunt activity is a PDF written in English - I don't know if it is available in other languages

### CODE REVIEW

File: csfieldguide/chapters/content/en/formal-languages/sections/finite-state-automata.md [hunk 1]
```diff
@@ -525,21 +527,23 @@ Then at the bottom of the window you can trace the string one character at a tim
...
 
 There are some FSA examples in the next section.
 If you're doing this for a report, keep copies of the automata and tests that you do (JFLAP's "File" menu has a "Save Image As..." option for taking snapshots of your work; alternatively you can save an FSA that you've created in a file to open later).
 
+{panel end}
+
 ### Examples to try
 
-Using Exorciser or JFLAP, construct an FSA that takes inputs made of the letters "a" and "b", and accepts the input if it meets one of the following requirements.
+By hand, or using Exorciser or JFLAP, construct an FSA that takes inputs made of the letters "a" and "b", and accepts the input if it meets one of the following requirements.
```
Comment (courtneycb) [inline csfieldguide/chapters/content/en/formal-languages/sections/finite-state-automata.md]: Consider changing to 'By hand, using Exorciser or JFLAP'

File: csfieldguide/chapters/content/structure/formal-languages/sections/sections.yaml [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 
 regular-expressions:
-  section-number: 1 #3
+  section-number: 3
 
 grammars-and-parsing:
-  section-number: 2 #4
+  section-number: 4
 
 the-whole-story:
```
Comment (courtneycb) [inline csfieldguide/chapters/content/structure/formal-languages/sections/sections.yaml]: The nfa guesser interactive in this section should uncheck the checkboxes when 'Generate new input sequence' is clicked.

File: csfieldguide/chapters/content/en/formal-languages/sections/finite-state-automata.md (hunk 1 above)
Comment (eAlasdair) [reply csfieldguide/chapters/content/en/formal-languages/sections/finite-state-automata.md]: That implies that you do the thing by hand with the tool Exorsier or JFLAP; doing what you suggest is more for something like to say `By hand, using a pen or pencil`

Comment (eAlasdair) [reply csfieldguide/chapters/content/en/formal-languages/sections/finite-state-automata.md]: But I will try to make it a bit nicer

**Pull request merged**